### PR TITLE
fix extraPortMappings / extraMounts and add more tests

### PR DESCRIPTION
**Pull request by BenTheElder**
- fix yaml unmarshal for types with custom unmarshal
- add test data to cover these

should be caught by CI next time. will spend more time on covering this in the future...

### CODE REVIEW

File: pkg/container/cri/yaml.go [hunk 1]
```diff
@@ -28,17 +28,24 @@ Custom YAML (de)serialization for these types
 // UnmarshalYAML implements custom decoding YAML
 // https://godoc.org/gopkg.in/yaml.v3
 func (m *Mount) UnmarshalYAML(unmarshal func(interface{}) error) error {
-	type Alias Mount
-	aux := &struct {
-		Propagation string `yaml:"propagation"`
-		*Alias
-	}{
```
Comment (BenTheElder) [inline pkg/container/cri/yaml.go]: sadly we have to give up this trick, yaml.v3 disallows duplicate keys.

File: pkg/container/cri/yaml.go [hunk 2]
```diff
@@ -28,17 +28,24 @@ Custom YAML (de)serialization for these types
...
-		Alias: (*Alias)(m),
+	// this is basically Mount, except Propogation is a string for further parsing
+	type MountYaml struct {
```
Comment (BenTheElder) [inline pkg/container/cri/yaml.go]: doing this gives pretty good errors though (type names show up in errors regarding keys)

File: pkg/container/cri/yaml.go (hunk 1 above)
Comment (aojea) [reply pkg/container/cri/yaml.go]: well, in the other said it forces us to be explicit :)

File: pkg/internal/apis/config/encoding/testdata/v1alpha3/invalid-bad-indent.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  - containerPath: /foo
+    hostPath: /bar
+    readOnly: true
+    selinuxRelabel: false
+    propagation: Bidirectional
+  extraPortMappings:
+  - containerPort: 8080
+    hostPort: 8080
+    protocol: UDP
+  - role: worker
```
Comment (BenTheElder) [inline pkg/internal/apis/config/encoding/testdata/v1alpha3/invalid-bad-indent.yaml]: we had an issue similar to this config, https://github.com/kubernetes-sigs/kind/issues/821

Comment (BenTheElder) [reply pkg/internal/apis/config/encoding/testdata/v1alpha3/invalid-bad-indent.yaml]: this is now an error :-)

File: pkg/internal/apis/config/encoding/testdata/v1alpha3/valid-many-fields.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+kind: Cluster
+apiVersion: kind.sigs.k8s.io/v1alpha3
+networking:
+  ipFamily: ipv6
```
Comment (BenTheElder) [inline pkg/internal/apis/config/encoding/testdata/v1alpha3/valid-many-fields.yaml]: doesn't cover _every_ field, however the mounts / port mappings are the finicky ones, will come back and add more test cases later.

**Comment by BenTheElder** [comment]
[gofmt-ed]

File: pkg/container/cri/yaml.go (hunk 1 above)
Comment (BenTheElder) [reply pkg/container/cri/yaml.go]: yeah. this was a clever and concise trick though, I had fun discovering this :-)

File: pkg/container/cri/yaml.go [hunk 5]
```diff
@@ -28,17 +28,24 @@ Custom YAML (de)serialization for these types
 // UnmarshalYAML implements custom decoding YAML
 // https://godoc.org/gopkg.in/yaml.v3
 func (m *Mount) UnmarshalYAML(unmarshal func(interface{}) error) error {
-	type Alias Mount
-	aux := &struct {
-		Propagation string `yaml:"propagation"`
-		*Alias
-	}{
-		Alias: (*Alias)(m),
+	// this is basically Mount, except Propogation is a string for further parsing
```
Comment (aojea) [inline pkg/container/cri/yaml.go]: nit typo Propogation -> Propagation

File: pkg/internal/apis/config/encoding/testdata/v1alpha3/valid-many-fields.yaml (hunk 4 above)
Comment (aojea) [reply pkg/internal/apis/config/encoding/testdata/v1alpha3/valid-many-fields.yaml]: I suggest creating an issue with help-wanted for these things, the same way you did the shellcheck meta issue in k/k

File: pkg/container/cri/yaml.go (hunk 5 above)
Comment (BenTheElder) [reply pkg/container/cri/yaml.go]: CI caught this too :P

Comment (BenTheElder) [reply pkg/container/cri/yaml.go]: done

**Comment by aojea** [comment]
/test pull-kind-conformance-parallel-1-16
failure unrelated to the PR

**Comment by BenTheElder** [comment]
prow failure specifically, no logs

File: pkg/internal/apis/config/encoding/testdata/v1alpha3/valid-many-fields.yaml (hunk 4 above)
Comment (aojea) [reply pkg/internal/apis/config/encoding/testdata/v1alpha3/valid-many-fields.yaml]: also, some fields like `ipFamily` are covered by validation, not perfect, but at least the bad config is not going through

**Comment by BenTheElder** [comment]
ignoring ipv6 flake

**Pull request merged**